docs/source/notes/cuda.rst

```diff
@@ -54,6 +54,26 @@ Below you can find a small example showcasing this::
         f = torch.randn(2).cuda(cuda2)
         # d.device, e.device, and f.device are all device(type='cuda', index=2)
 
+.. _tf32_on_ampere:
+
+TensorFloat-32(TF32) on Ampere devices
+--------------------------------------
+
+Starting from `CUDA 11`_ and the `Ampere architecture`_, it is possible to compute
```

Comment:
This paragraph needs work. In fact, it should probably be expanded into several paragraphs.

- It needs to define its terms before using them. For example, what is a "math mode?" 
- It needs to quantify what "sacrificing precision" means, and what kind of a speedup users can expect and when. The current language around the "10x speedup" sounds like marketing, and "sacrificing precision" is frighteningly vague.
- It needs an example.

Here's a possible restructuring:

- Starting in PyTorch X.X, there's a new flag, Y, that's on by default(?). This flag determines where CUDA GEMMs and convolutions with float32 input tensors use TensorFloat32 tensor cores internally
- These cores compute X with Y throughput, compared with performing the operation entirely in float32
- There may be a loss of precision when doing this (why?, and can you quantify the loss of precision?)
- Example code performing a gemm/convolution with the flag enabled/disabled, you can describe perf and compare the values to demonstrate the performance difference / precision difference
- Discuss when a user may want to disable the flag - do you have an example for this, too?